src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultIEnumerableConstructibleConverter.cs

```diff
@@ -15,8 +15,13 @@ internal sealed class DefaultIEnumerableConstructibleConverter : JsonEnumerableC
     {
         public override IEnumerable CreateFromList(Type enumerableType, Type elementType, IList sourceList)
         {
-            Array array = CreateArrayFromList(elementType, sourceList);
-            return (IEnumerable)Activator.CreateInstance(enumerableType, array);
+            // TODO: Cache reflection here, or modify JsonPropertyInfoCommon to have a TElementType generic parameter.
```

Comment:
The JsonPropertyInfoCommon does have a TElementType - but you have to get the "property policy" first, like this code that calls `GetDictionaryConcreteType`:
`jsonInfo.ElementClassInfo.GetPolicyProperty().GetDictionaryConcreteType();`